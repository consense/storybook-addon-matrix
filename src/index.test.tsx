import * as React from 'react';
import { mount } from 'enzyme';
import { withMatrix } from '.';

type BaseComponentProps = {
  title?: string;
  text: string;
};

const BaseComponent: React.FC<BaseComponentProps> = ({ title, text }) => (
  <div>
    {title && <h1>{title}</h1>}
    <p>{text}</p>
  </div>
);

const TestComponent: React.FC = () => <BaseComponent title="TITLE" text="TEXT" />;

describe('addon Matrix', () => {
  const createStoryFn = (Component: React.ElementType) => () => <Component />;
  const storyFn = createStoryFn(TestComponent);
  it('should render <Matrix />', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        component: BaseComponent,
        matrix: {
          pattern: {
            title: [undefined, 'title pattern 1'],
            text: ['text pattern 1', 'text pattern 2'],
          },
        },
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });
});
