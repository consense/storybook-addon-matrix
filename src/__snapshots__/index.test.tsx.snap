// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addon Matrix should render <Error /> when component and pattern are not exist 1`] = `
<Component
  messages={
    Array [
      "Default export component must be present",
      "Parameter pattern must be present",
    ]
  }
>
  <styled.div
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="sc-AxjAm bwuNzm"
      color="white"
    >
      <styled.span
        fontSize={2}
      >
        <span
          className="sc-AxirZ bCWNZw"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </styled.span>
      <ul>
        <li
          key="Default export component must be present"
        >
          <styled.span>
            <span
              className="sc-AxirZ HanLJ"
            >
              Default export component must be present
            </span>
          </styled.span>
        </li>
        <li
          key="Parameter pattern must be present"
        >
          <styled.span>
            <span
              className="sc-AxirZ HanLJ"
            >
              Parameter pattern must be present
            </span>
          </styled.span>
        </li>
      </ul>
    </div>
  </styled.div>
</Component>
`;

exports[`addon Matrix should render <Error /> when component is not exist 1`] = `
<Component
  messages={
    Array [
      "Default export component must be present",
    ]
  }
>
  <styled.div
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="sc-AxjAm bwuNzm"
      color="white"
    >
      <styled.span
        fontSize={2}
      >
        <span
          className="sc-AxirZ bCWNZw"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </styled.span>
      <ul>
        <li
          key="Default export component must be present"
        >
          <styled.span>
            <span
              className="sc-AxirZ HanLJ"
            >
              Default export component must be present
            </span>
          </styled.span>
        </li>
      </ul>
    </div>
  </styled.div>
</Component>
`;

exports[`addon Matrix should render <Error /> when pattern is not exist 1`] = `
<Component
  messages={
    Array [
      "Parameter pattern must be present",
    ]
  }
>
  <styled.div
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="sc-AxjAm bwuNzm"
      color="white"
    >
      <styled.span
        fontSize={2}
      >
        <span
          className="sc-AxirZ bCWNZw"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </styled.span>
      <ul>
        <li
          key="Parameter pattern must be present"
        >
          <styled.span>
            <span
              className="sc-AxirZ HanLJ"
            >
              Parameter pattern must be present
            </span>
          </styled.span>
        </li>
      </ul>
    </div>
  </styled.div>
</Component>
`;

exports[`addon Matrix should render <Matrix /> 1`] = `
Array [
  <styled.div>
    <div
      className="sc-AxjAm dYfHlB"
    >
      <TestComponent>
        <BaseComponent
          text="TEXT"
          title="TITLE"
        >
          <div>
            <h1>
              TITLE
            </h1>
            <p>
              TEXT
            </p>
          </div>
        </BaseComponent>
      </TestComponent>
    </div>
  </styled.div>,
  <Component
    component={[Function]}
    originalProps={Object {}}
    propsPattern={
      Object {
        "text": Array [
          "text pattern 1",
          "text pattern 2",
        ],
        "title": Array [
          undefined,
          "title pattern 1",
        ],
      }
    }
  >
    <styled.div
      border="1px solid #ddd"
      borderRadius={3}
      boxShadow="1px 1px 6px #eee"
      key="%7B%22text%22%3A%22text%20pattern%201%22%7D"
      m={2}
      mb={3}
    >
      <div
        className="sc-AxjAm cgWUlu"
      >
        <styled.div
          bg="#eeeeee"
          p={2}
        >
          <div
            className="sc-AxjAm Zfmpp"
          >
            <styled.span>
              <span
                className="sc-AxirZ HanLJ"
              >
                {"text":"text pattern 1"}
              </span>
            </styled.span>
          </div>
        </styled.div>
        <styled.div
          m={2}
        >
          <div
            className="sc-AxjAm brQMOi"
          >
            <BaseComponent
              text="text pattern 1"
            >
              <div>
                <p>
                  text pattern 1
                </p>
              </div>
            </BaseComponent>
          </div>
        </styled.div>
      </div>
    </styled.div>
    <styled.div
      border="1px solid #ddd"
      borderRadius={3}
      boxShadow="1px 1px 6px #eee"
      key="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%201%22%7D"
      m={2}
      mb={3}
    >
      <div
        className="sc-AxjAm cgWUlu"
      >
        <styled.div
          bg="#eeeeee"
          p={2}
        >
          <div
            className="sc-AxjAm Zfmpp"
          >
            <styled.span>
              <span
                className="sc-AxirZ HanLJ"
              >
                {"title":"title pattern 1","text":"text pattern 1"}
              </span>
            </styled.span>
          </div>
        </styled.div>
        <styled.div
          m={2}
        >
          <div
            className="sc-AxjAm brQMOi"
          >
            <BaseComponent
              text="text pattern 1"
              title="title pattern 1"
            >
              <div>
                <h1>
                  title pattern 1
                </h1>
                <p>
                  text pattern 1
                </p>
              </div>
            </BaseComponent>
          </div>
        </styled.div>
      </div>
    </styled.div>
    <styled.div
      border="1px solid #ddd"
      borderRadius={3}
      boxShadow="1px 1px 6px #eee"
      key="%7B%22text%22%3A%22text%20pattern%202%22%7D"
      m={2}
      mb={3}
    >
      <div
        className="sc-AxjAm cgWUlu"
      >
        <styled.div
          bg="#eeeeee"
          p={2}
        >
          <div
            className="sc-AxjAm Zfmpp"
          >
            <styled.span>
              <span
                className="sc-AxirZ HanLJ"
              >
                {"text":"text pattern 2"}
              </span>
            </styled.span>
          </div>
        </styled.div>
        <styled.div
          m={2}
        >
          <div
            className="sc-AxjAm brQMOi"
          >
            <BaseComponent
              text="text pattern 2"
            >
              <div>
                <p>
                  text pattern 2
                </p>
              </div>
            </BaseComponent>
          </div>
        </styled.div>
      </div>
    </styled.div>
    <styled.div
      border="1px solid #ddd"
      borderRadius={3}
      boxShadow="1px 1px 6px #eee"
      key="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%202%22%7D"
      m={2}
      mb={3}
    >
      <div
        className="sc-AxjAm cgWUlu"
      >
        <styled.div
          bg="#eeeeee"
          p={2}
        >
          <div
            className="sc-AxjAm Zfmpp"
          >
            <styled.span>
              <span
                className="sc-AxirZ HanLJ"
              >
                {"title":"title pattern 1","text":"text pattern 2"}
              </span>
            </styled.span>
          </div>
        </styled.div>
        <styled.div
          m={2}
        >
          <div
            className="sc-AxjAm brQMOi"
          >
            <BaseComponent
              text="text pattern 2"
              title="title pattern 1"
            >
              <div>
                <h1>
                  title pattern 1
                </h1>
                <p>
                  text pattern 2
                </p>
              </div>
            </BaseComponent>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Component>,
]
`;
